src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpReferencesTests.cs

```diff
@@ -28,9 +28,6 @@ public void AddingReferenceToProjectMetadataPromotesToProjectReference()
                 project2.OnImportAdded(@"c:\project1.dll", "project1");
 
                 Assert.Equal(true, project2.GetCurrentProjectReferences().Any(pr => pr.ProjectId == project1.Id));
-
-                project2.Disconnect();
```

Comment:
Yes, the TestEnvironment was already calling Disconnect for all projects.  These lines caused Disconnect to be called multiple times on the same instance, which eventually lead to exceptions due to disposed objects being re-disposed.  When the project remove was immediate, this was circumvented somehow.